memory_statistics/memory_statistics.py

```diff
@@ -127,6 +131,10 @@ def parse_arguments():
 def main():
     args = parse_arguments()
 
+    if not shutil.which("arm-none-eabi-gcc"):
+        print("ARM GCC not found. Please add the ARM GCC toolchain to your path.")
+        sys.exit(1)
```

Comment:
Is it possible that different versions of this toolchain result in different estimates? If yes, we probably need to verify the version as well.

Reply:
Yes, but using other versions needs to be allowed for testing new versions or comparing. This message is mostly for convenience when running locally, so you get a readable error when you forget to add the toolchain to your path. The correct version will be enforced by the PR checks